server/routes/auth-routes.js

```diff
@@ -0,0 +1,60 @@
+const router = require('express').Router();
+const passport = require('passport');
+let client_add = "http://localhost:3000";
+
+// auth login
+router.get('/login', (req,res) => {
+  res.render('login');
+});
+
+// auth logout
+router.get('/logout', (req,res) => {
+    if (req.user !== null) {
+        req.logout();
+        res.send('logging out');
+    }
+  // handle with passport
+});
+
+// auth with our own Login
+router.get('/local', (req, res) => {
+  res.render('ourlogin');
+});
+
+// auth with our own signup
+router.get('/signup', (req, res) => {
+  res.render('signup');
+});
+
+// auth with google
+router.get('/google', passport.authenticate('google', {
+  scope: ['profile', 'email']
+}));
+
+// callback route for google to redirect to
+router.get('/google/redirect',
```

Comment:
passport.authenticate() does the initial login, handle subsequent authentication by checking req.isAuthenticated(). This way, in building out the other api routes, we can use one authentication checker. see https://github.com/JumboCode/E4P/blob/2c7c7a5129d011bb599460ab7c2fffba09065fb4/routes/adminRoutes.js#L14

Reply:
Yes, I will be using this method in other rules. For google login, the authenticate has two phases, 1. redirecting to google, and 2. receive info (callback URL) from google. I believe both phases require passport.authenticate(), as is in the official docs/examples. `http://www.passportjs.org/docs/google/`